src/crypto.ts

```diff
@@ -226,37 +228,49 @@ export default class Crypto {
       }
     })
 
-    const downloadPromises : Array<Promise<void>> = []
-    for (let fieldId in attachmentRecords) {
-      // Original name for the file is not found
-      if (filenames[fieldId] === undefined) return null
-
-      downloadPromises.push(
-        axios.get(attachmentRecords[fieldId], { responseType: 'json' })
-      .then((downloadResponse) => {
-        const encryptedAttachment: EncryptedAttachmentContent = downloadResponse.data
-        const encryptedFile: EncryptedFileContent = {
-          submissionPublicKey: encryptedAttachment.encryptedFile.submissionPublicKey,
-          nonce: encryptedAttachment.encryptedFile.nonce,
-          binary: decodeBase64(encryptedAttachment.encryptedFile.binary),
-        }
-
-        return this.decryptFile(formSecretKey, encryptedFile)
-      }).then((decryptedFile) => {
-        if (decryptedFile === null) throw new Error("Attachment decryption failed")
-        decryptedRecords[fieldId] = { filename: filenames[fieldId], content: decryptedFile }
-      }))
+    const fieldIds = Object.keys(attachmentRecords)
+    // Check if all fieldIds are within filenames
+    if (!areAttachmentFieldIdsValid(fieldIds, filenames)) {
+      return null
     }
 
+    const downloadPromises = fieldIds.map((fieldId) => {
+      return (
+        axios
+          // Retrieve all the attachments as JSON
+          .get(attachmentRecords[fieldId], { responseType: 'json' })
+          // Decrypt all the attachments
+          .then((downloadResponse) => {
+            const encryptedAttachment: EncryptedAttachmentContent =
+              downloadResponse.data
+            const encryptedFile: EncryptedFileContent = {
+              submissionPublicKey:
+                encryptedAttachment.encryptedFile.submissionPublicKey,
+              nonce: encryptedAttachment.encryptedFile.nonce,
+              binary: decodeBase64(encryptedAttachment.encryptedFile.binary),
+            }
+            return this.decryptFile(formSecretKey, encryptedFile)
+          })
+          .then((decryptedFile) => {
+            if (decryptedFile === null)
+              throw new Error('Attachment decryption failed')
```

Comment:
can use a custom error here?

Reply:
this is meant to be caught internally right? regardless of what kind of error is thrown, we return `null`

Reply:
i wrote a custom error at the start but i removed it because my understanding is that our library treats `errors` not as values but as exceptions, and in that sense, should never throw (as much as possible). 

will change if needed!